hittingEndPoint.vb

```diff
@@ -0,0 +1,176 @@
+        oauth2.AuthorizationEndpoint = "https://accounts.google.com/o/oauth2/auth"
+        oauth2.TokenEndpoint = "https://oauth2.googleapis.com/token"
+        'TODO: set your "clientId" and clientSecretId".
+        oauth2.ClientId = userClientId
+        oauth2.ClientSecret = userClientSecretId
+        oauth2.CodeChallenge = True
+        oauth2.CodeChallengeMethod = "S256"
+        oauth2.Scope = "https://www.googleapis.com/auth/spreadsheets"
+        Dim url As String = oauth2.StartAuth()
+        If (oauth2.LastMethodSuccess <> True) Then
+            MessageBox.Show(oauth2.LastErrorText)
+            Return ""
+        End If
+        System.Diagnostics.Process.Start(url)
+        Dim numMsWaited As Integer = 0
+        While (numMsWaited < 30000) And (oauth2.AuthFlowState < 3)
+            oauth2.SleepMs(100)
+            numMsWaited = numMsWaited + 100
+        End While
+
+        If (oauth2.AuthFlowState < 3) Then
+            oauth2.Cancel()
+            MessageBox.Show("No response from the browser!")
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 5) Then
+            MessageBox.Show("OAuth2 failed to complete.")
+            MessageBox.Show(oauth2.FailureInfo)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState = 4) Then
+            MessageBox.Show("OAuth2 authorization was denied.")
+            MessageBox.Show(oauth2.AccessTokenResponse)
+            Return ""
+        End If
+        If (oauth2.AuthFlowState <> 3) Then
+            MessageBox.Show("Unexpected AuthFlowState:" & oauth2.AuthFlowState)
+            Return ""
+        End If
+        Return oauth2.AccessToken
+    End Function
+
+    Public Function callSheetsAPI(ByRef fileId As String, ByRef userClientId As String , ByRef userClientSecretId As String) As List(Of String)
+        Dim lines As List(Of String) = Nothing
+        'Calls for Authorization Token.
+        Dim _bearerToken As String = getAuthorizationToken(userClientId,userClientSecretId)
+        If _bearerToken <> "" Then
+            Dim myUri As New Uri("https://docs.google.com/spreadsheets/vbaprocessfile?fid=" + fileId)
+            Dim myWebRequest = System.Net.HttpWebRequest.Create(myUri)
+            Dim myHttpWebRequest = CType(myWebRequest, System.Net.HttpWebRequest)
+            myHttpWebRequest.Method = "GET"
+            myHttpWebRequest.PreAuthenticate = True
+            myHttpWebRequest.Headers.Add("Authorization", "Bearer " & _bearerToken)
+            Dim myWebResponse As HttpWebResponse = myWebRequest.GetResponse()
+            'Collectting the response after request.
+            Dim responseStream As Stream = myWebResponse.GetResponseStream()
+            If responseStream Is Nothing Then
+                MessageBox.Show("Api didn't respond.")
+                Return lines
+            End If
+            'Creating a file to store data from the response stream.
+            Dim Json As New FileStream("D:\Json.txt", FileMode.Create)

```

Comment:
Add a TODO to handle other operating systems.

Reply:
Done.